### PR TITLE
SQL Trigger functions for last_updated attribute on Projects and Tasks

### LINKED ISSUE #75: SQL Trigger Functions
Write SQL trigger functions into our database to perform common tasks such as updating the `last_updated` property of the Project records whenever an associated sprint/user/task/comment is updated/created/deleted.

This is a stretch goal, and as such it is not necessary for the completion of this project.

### PR DESCRIPTION
**Pull request by barreryu**
Closes #75

### CODE REVIEW

File: database/RecreateOPMDatabase.sql [hunk 1]
```diff
@@ -119,3 +119,91 @@ CREATE TABLE Comments (
...
+
+CREATE TRIGGER trigger_update_tasks_project_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Tasks
+FOR EACH ROW 
+EXECUTE FUNCTION update_project_last_updated();
+
+-- PROJECTS
+-- Updating Project's last_updated when Project is CRUD
+CREATE TRIGGER trigger_update_project_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Projects
```
Comment (GPropersi) [inline database/RecreateOPMDatabase.sql:180]: Does this mean the `last_updated` for Projects will be updated when it's about to be deleted?

File: database/RecreateOPMDatabase.sql [hunk 2]
```diff
@@ -119,3 +119,91 @@ CREATE TABLE Comments (
...
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+
+--TRIGGERS
+-- TASKS
+-- Updating Task's last_updated along with Task's associated Projects last_updated when task is CRUD
+CREATE TRIGGER trigger_update_task_last_updated
+BEFORE INSERT OR UPDATE OR DELETE ON Tasks
```
Comment (GPropersi) [inline database/RecreateOPMDatabase.sql:168]: Similar Q, does this mean the Task `last_updated` will be updated before it's deleted?

File: database/RecreateOPMDatabase.sql [hunk 3]
```diff
@@ -119,3 +119,91 @@ CREATE TABLE Comments (
...
+$$ LANGUAGE plpgsql;
+
+-- Updates last_updated of called for Table
+CREATE OR REPLACE FUNCTION update_last_updated()
+RETURNS TRIGGER AS $$
+BEGIN
+    NEW.last_updated = CURRENT_TIMESTAMP;
+    RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
```
Comment (GPropersi) [inline database/RecreateOPMDatabase.sql:148]: What's the difference between these three? Do we just need the first two? I'm totally out of the loop on trigger functions, not sure what the third one is doing vs the first two.

**Review by GPropersi — CHANGES_REQUESTED** [review]
These look great Ryu, nice work! I had some general questions I left in comments. I noticed your commit also mentions you changed some Java code, but I'm not seeing the code in the PR - did you need to make another push of your code?

File: database/RecreateOPMDatabase.sql (hunk 2 above)
Comment (barreryu) [reply database/RecreateOPMDatabase.sql:168]: Answering this question and the one above, but yes. We can also set it to AFTER if you'd like, but I heard BEFORE is better when we have exceptions to data entry. Not sure if we have any of that though

File: database/RecreateOPMDatabase.sql (hunk 3 above)
Comment (barreryu) [reply database/RecreateOPMDatabase.sql:148]: For some reason calling the task specific function for updating the task threw an error, similar with the project function when triggering the project. I think something about how SQL doesn't like it when a table's trigger references a function that is also referencing the same table? Not too sure though
The third one is used for those two cases where task is updating itself and the project is updating itself.

File: database/RecreateOPMDatabase.sql (hunk 2 above)
Comment (GPropersi) [reply database/RecreateOPMDatabase.sql:168]: Do we want to update their `last_updated` before they get deleted?

Comment (barreryu) [reply database/RecreateOPMDatabase.sql:168]: I'll change to AFTER

**Review by GPropersi — APPROVED** [review]

**Pull request merged**